pkg/db/db_session/default.go

```diff
@@ -189,12 +194,17 @@ func newListener(ctx context.Context, dbConfig *config.DatabaseConfig, channel s
 		panic(err)
 	}
 
-	logger.Infof("Starting channeling monitor for %s", channel)
-	waitForNotification(ctx, listener, callback)
+	return listener
 }
 
-func (f *Default) NewListener(ctx context.Context, channel string, callback func(id string)) {
-	newListener(ctx, f.config, channel, callback)
+func (f *Default) NewListener(ctx context.Context, channel string, callback func(id string)) *pq.Listener {
+	logger := ocmlogger.NewOCMLogger(ctx)
+
+	listener := newListener(ctx, f.config, channel)
```

Comment:
how about new the listener in the goroutine waitForNotification? we can maintain the listener in one place, may like
```
func waitForNotification() {
   listener := newListener(ctx, f.config, channel)
   for {
     		// ...	
		// recreate the listener
		listener = newListener(ctx, dbConfig, channel)
        }
    }
}
```

Reply:
if encapsulate the listener into `waitForNotification`, we cannot get the listener in the integration test. without the listener we cannot close the listener to generate an errListenerClosed to test the listener re-creation.

Reply:
ok ...

may add some comments, the returned  listener only for test purpose, it cannot be shared by others?